src/core/net/ip6_address.hpp

```diff
@@ -269,6 +272,24 @@ class Address : public otIp6Address
      */
     bool IsIidReserved(void) const;
 
+    /**
+     * This method gets the IPv6 address locator.
+     *
+     * @returns RLOC16 or ALOC16.
+     *
+     */
+    inline uint16_t GetLocator(void) const { return HostSwap16(mFields.m16[7]); }
+
+    /**
+     * This method sets the IPv6 address locator.
+     *
+     * This method only changes the last 2 bytes of the address and keeps the rest of the address as before.
+     *
+     * @param[in]  aLocator   RLOC16 or ALOC16.
+     *
+     */
+    inline void SetLocator(uint16_t aLocator) { mFields.m16[7] = HostSwap16(aLocator); }
```

Comment:
```suggestion
    void SetLocator(uint16_t aLocator) { mFields.m16[7] = HostSwap16(aLocator); }
```
No need to explicitly declare as inline.